python/cudf/cudf/core/column/timedelta.py

```diff
@@ -484,88 +484,57 @@ def components(self) -> dict[str, ColumnBase]:
         4     37     13       12       14           234             0            0
         """
 
+        @functools.cache
```

Comment:
nitpick: `from functools import cache, cached_property`
```suggestion
        @cache
```

Reply:
I'm OK either way

Reply:
I just left this as `functools.cache` to leave the existing import as is